hamt/util.go

```diff
@@ -61,8 +75,74 @@ func logtwo(v int) (int, error) {
 	return lg2, nil
 }
 
-func hash(val []byte) []byte {
+func murmur3Hash(val []byte) []byte {
 	h := murmur3.New64()
 	h.Write(val)
 	return h.Sum(nil)
 }
+
+// ONLY FOR TESTING: Return the same value as the hash.
+func IdHash(val []byte) []byte {
+	return val
+}
+
+// CreateCompleteHAMT creates a HAMT the following properties:
+// * its height (distance/edges from root to deepest node) is specified by treeHeight.
+// * all leaf Shard nodes have the same depth (and have only 'value' links).
+// * all internal Shard nodes point only to other Shards (and hence have zero 'value' links).
+// * the total number of 'value' links (directory entries) is:
+//   io.DefaultShardWidth ^ treeHeight.
+// FIXME: HAMTHashFunction needs to be set to IdHash by the caller. We depend on
+//  this simplification for the current logic to work. (HAMTHashFunction is a
+//  global setting of the package, it is hard-coded in the serialized Shard node
+//  and not allowed to be changed on a per HAMT/Shard basis.)
+//  (If we didn't rehash inside setValue then we could just generate
+//  the fake hash as in io.SetAndPrevious through `newHashBits()` and pass
+//  it as an argument making the hash independent of tree manipulation; that
+//  sounds as the correct way to go in general and we wouldn't need this.)
+func CreateCompleteHAMT(ds ipld.DAGService, treeHeight int) (ipld.Node, error) {
```

Comment:
functions only used by go tests should live in _test.go files if possible.

Note: if you're running into dependency cycle issues or issues with breaking the API a tool that may be your friend here is the `internal` package which allows things to be exported for use within the module, but not outside of it.